server/gui/main.py

```diff
@@ -0,0 +1,212 @@
+# flake8: noqa F403, F405
+import sys
+from os.path import splitext, basename
+
+from cefpython3 import cefpython as cef
+from PyQt5.QtCore import *
+from PyQt5.QtWidgets import *
+
+from server.gui.browser import CefWidget, CefApplication
+from server.gui.cxg_server import cellxgeneServer, DataLoadWorker, ServerRunWorker
+from server.gui.utils import WINDOWS, LINUX, MAC
+
+# Configuration
+# TODO remember this or calculate it?
+WIDTH = 1024
+HEIGHT = 768
+
+
+# noinspection PyUnresolvedReferences
+class MainWindow(QMainWindow):
+    def __init__(self):
+        super(MainWindow, self).__init__(None)
+        self.thread_pool = QThreadPool()
+        self.cef_widget = None
+        self.data_widget = None
+        self.server = cellxgeneServer(self)
+        self.server.setup_app()
+        self.runServer()
+        self.setWindowTitle("cellxgene")
+
+        # Strong focus - accepts focus by tab & click
+        self.setFocusPolicy(Qt.StrongFocus)
+        self.setupLayout()
+        self.setupMenu()
+
+    def setupLayout(self):
+        self.resize(WIDTH, HEIGHT)
+        self.cef_widget = CefWidget(self)
+        self.data_widget = LoadWidget(self)
+        self.stacked_layout = QStackedLayout()
+        self.stacked_layout.addWidget(self.data_widget)
+        self.stacked_layout.addWidget(self.cef_widget)
+        main_layout = QVBoxLayout()
+        main_layout.setContentsMargins(0, 0, 0, 0)
+        main_layout.setSpacing(0)
+        main_layout.addLayout(self.stacked_layout)
+        frame = QFrame()
+        frame.setLayout(main_layout)
+        self.setCentralWidget(frame)
+
+        if WINDOWS:
+            # On Windows with PyQt5 main window must be shown first
+            # before CEF browser is embedded, otherwise window is
+            # not resized and application hangs during resize.
+            self.show()
+
+        # Browser can be embedded only after layout was set up
+        self.cef_widget.embedBrowser()
+
+        if LINUX:
+            # On Linux with PyQt5 the QX11EmbedContainer widget is
+            # no more available. An equivalent in Qt5 is to create
```

Comment:
"no more available"?  maybe "no longer available" or "not available"?

Reply:
LOL! This was directly taken from the cefpython demo, I guess I forgot to copy edit. I'll fix it.